### PR TITLE
feat: show filename when throwing

### PR DESCRIPTION
**Pull request by saschanaz**
For https://github.com/w3c/respec/issues/2235.

### CODE REVIEW

**Comment by saschanaz** [comment]
Ah, I have to think more about the multifile cases...

**Comment by saschanaz** [comment]
Just add support for array input. Might just be okay.

File: lib/error.js [hunk 1]
```diff
@@ -50,10 +50,12 @@ function error(source, position, current, message, type) {
   const contextualMessage = precedingLine + sobsequentLine + "\n" + spaced;
 
   const contextType = type === "Syntax" ? "since" : "inside";
+  const fileContext = source.fileName ? ` in ${source.fileName}` : "";
```
Comment (marcoscaceres) [inline lib/error.js]: Super annoying nit: filename is just one word in English 🥺

Comment (marcoscaceres) [inline lib/error.js]:
```suggestion
  const fileContext = source.filename ? ` in ${source.filename}` : "";
```

File: lib/tokeniser.js [hunk 2]
```diff
@@ -223,10 +223,11 @@ export class Tokeniser {
 }
 
 class WebIDLParseError extends Error {
-  constructor({ message, line, input, tokens }) {
+  constructor({ message, line, fileName, input, tokens }) {
```
Comment (marcoscaceres) [inline lib/tokeniser.js]:
```suggestion
  constructor({ message, line, filename, input, tokens }) {
```

File: lib/tokeniser.js [hunk 3]
```diff
@@ -223,10 +223,11 @@ export class Tokeniser {
...
     super(message);
     this.name = "WebIDLParseError"; // not to be mangled
     this.line = line;
+    this.fileName = fileName;
```
Comment (marcoscaceres) [inline lib/tokeniser.js]:
```suggestion
    this.filename = filename;
```

File: lib/error.js (hunk 1 above)
Comment (saschanaz) [reply lib/error.js]: nit 2: it's not always for a file. What would be a better name? `trackingName`? Just `name`?

File: lib/webidl2.js [hunk 4]
```diff
@@ -92,5 +92,8 @@ function parseByTokens(tokeniser, options) {
 
 export function parse(str, options = {}) {
   const tokeniser = new Tokeniser(str);
+  if (options.fileName) {
```
Comment (marcoscaceres) [inline lib/webidl2.js]:
```suggestion
  if (options.filename) {
```

File: lib/error.js (hunk 1 above)
Comment (marcoscaceres) [reply lib/error.js]: `name` could work :)

Comment (marcoscaceres) [reply lib/error.js]: I'll delete my suggestions.

**Comment by saschanaz** [comment]
>will we need to generate a name?

IMO yes, I'll probably just assign the index of `<pre>` array and use it to point the right elements.

**Comment by marcoscaceres** [comment]
> IMO yes,

Sounds good. We can work that out on the ReSpec side.

**Comment by saschanaz** [comment]
Work on it yesterday and failed to do xref thing there. Will open a PR without xref soon.

**Comment by saschanaz** [comment]
Had to use `sourceName` instead because `name` in error.js conflicts with `Error.name`. Does it look good?

**Comment by marcoscaceres** [comment]
sourceName is good too :)

**Comment by saschanaz** [comment]
(Waiting on your approval)

**Pull request merged**